salt/minion.py

```diff
@@ -198,6 +199,20 @@ def resolve_dns(opts, fallback=True):
     return ret
 
 
+def srv_lookup(name, protocol='TCP', domain=None):
+    if domain:
+        domain = '.' + domain
+    record_name = '_{0}_._{1}{2}'.format(name, protocol.lower(), domain or '')
+    cmd = ['dig', '+short', record_name, 'SRV']
```

Comment:
host equiv would be:

cmd = ['host', '-t', 'SRV', record_name]

The regex seems to work fine when using this.